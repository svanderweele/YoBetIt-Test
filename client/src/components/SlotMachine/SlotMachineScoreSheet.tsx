import React from "react";
import { Table, Card } from "react-bootstrap";
import {
  SlotMachineRewardRequirement,
} from "../../models/SlotMachine";

interface SlotMachineScoreSheetProps {
  rewardRequirements: SlotMachineRewardRequirement[];
}

const SlotMachineScoreSheet = (props: SlotMachineScoreSheetProps) => {
  return (
    <Card>
      <Card.Body>
        <Card.Title>Slot Machine Score Sheet</Card.Title>
        <Table bordered>
          <thead>
            <tr>
              <th>Id</th>
              <th>Pattern</th>
              <th>Reward</th>
            </tr>
          </thead>
          <tbody>
            {props.rewardRequirements.map(
              (reward: SlotMachineRewardRequirement) => {
                return (
                  <tr key={reward.id}>
                    <td>{reward.id}</td>
                    <td>{reward.patternString}</td>
                    <td>{reward.reward}</td>
                  </tr>
                );
              }
            )}
          </tbody>
        </Table>
      </Card.Body>
    </Card>
  );
};

export default SlotMachineScoreSheet;
