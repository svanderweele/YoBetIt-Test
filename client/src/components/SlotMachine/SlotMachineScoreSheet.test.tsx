import React from "react";
import { shallow } from "enzyme";
import SlotMachineScoreSheet from "./SlotMachineScoreSheet";
import {
  SlotMachineRewardRequirement,
  SlotMachinePatternTypeEnum,
} from "../../models/SlotMachine";
describe("Slot Machine Score Sheet", () => {
  it("should render correctly", () => {
    const requirements = [
      new SlotMachineRewardRequirement(
        [
          SlotMachinePatternTypeEnum.Apple,
          SlotMachinePatternTypeEnum.Apple,
          SlotMachinePatternTypeEnum.Apple,
        ],
        20
      ),
    ];

    requirements.forEach((requirements, index) => (requirements.id = index));

    const component = shallow(
      <SlotMachineScoreSheet rewardRequirements={requirements} />
    );
    expect(component).toMatchSnapshot();
  });
});
