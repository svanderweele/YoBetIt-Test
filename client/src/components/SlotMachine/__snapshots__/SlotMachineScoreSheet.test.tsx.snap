// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot Machine Score Sheet should render correctly 1`] = `
<Card
  body={false}
>
  <CardBody>
    <CardTitle>
      Slot Machine Score Sheet
    </CardTitle>
    <ForwardRef
      bordered={true}
    >
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Pattern
          </th>
          <th>
            Reward
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="0"
        >
          <td>
            0
          </td>
          <td />
          <td>
            20
          </td>
        </tr>
      </tbody>
    </ForwardRef>
  </CardBody>
</Card>
`;
